src/zenml/cli/secret.py

```diff
@@ -69,195 +61,292 @@ def secret(ctx: click.Context) -> None:
+                        f"Please enter the secret_value " f"for the key [{k}]:"
+                    )
+                else:
+                    warning(
+                        f"Key {k} already in this secret. Please restart this"
+                        f" process or use "
+                        f"'zenml secret update {name} --{secret_schema_type}'"
+                        f" to update this key after the secret is registered. "
+                        f"Skipping ..."
+                    )
+
+                if not click.confirm(
+                    "Do you want to add another key-value pair to this secret?"
+                ):
+                    break
 
     else:
-        click.echo(
-            "You have not supplied a secret_set_schema with any "
-            "predefined keys. Entering interactive mode:"
-        )
-        while True:
-            k = click.prompt("Please enter a secret-key")
-            if k not in secret_contents:
-                secret_contents[k] = getpass.getpass(
-                    f"Please enter the secret_value " f"for the key [{k}]:"
-                )
-            else:
-                warning(
-                    f"Key {k} already in this secret. Please restart this"
-                    f" process or use "
-                    f"'zenml secret update {name} --{secret_schema_type}'"
-                    f" to update this key. "
-                    f"Skipping ..."
-                )
-
-            if not click.confirm(
-                "Do you want to add another key-value pair to this secret?"
-            ):
-                break
+        if not parsed_args:
+            error(
+                "Secret fields must be passed as arguments when not using "
+                "interactive mode."
+            )
+
+        secret_contents.update(parsed_args)
+
+    try:
+        secret = secret_schema(**secret_contents)
+    except ValidationError as e:
+        error(f"Secret values do not conform with the secret schema: {str(e)}")
 
     click.echo("The following secret will be registered.")
-    secret = secret_schema(**secret_contents)
     pretty_print_secret(secret=secret, hide_secret=True)
 
-    with console.status(f"Saving secret set `{name}`..."):
+    with console.status(f"Saving secret `{name}`..."):
         secrets_manager.register_secret(secret=secret)
 
 
-@secret.command("get", help="Get a secret by its name.")
+@secret.command("get")
 @click.argument("name", type=click.STRING)
 @click.pass_obj
 def get_secret(
     secrets_manager: "BaseSecretsManager",
     name: str,
 ) -> None:
-    """Get a secret set, given its name.
-
-    Args:
-        secrets_manager: Stack component that implements the interface to the
-            underlying secrets engine
-        name: Name of the secret
-    """
-    # with console.status(f"Getting secret set `{name}`..."):
+    """Get a secret, given its name."""
+    # with console.status(f"Getting secret `{name}`..."):
```

Comment:
Should we uncomment or remove the console status here?

Reply:
remove it is